src/Serilog.Sinks.Seq/Sinks/Seq/HttpLogShipper.cs

```diff
@@ -280,6 +282,70 @@ void OnTick()
             }
         }
 
+        const string InvalidPayloadFilePrefix = "invalid-";
+        async Task DumpInvalidPayload(HttpResponseMessage result, string payload)
+        {
+            var invalidPayloadFilename = $"{InvalidPayloadFilePrefix}{result.StatusCode}-{Guid.NewGuid():n}.json";
+            var invalidPayloadFile = Path.Combine(_logFolder, invalidPayloadFilename);
+            var resultContent = await result.Content.ReadAsStringAsync();
+            SelfLog.WriteLine("HTTP shipping failed with {0}: {1}; dumping payload to {2}", result.StatusCode, resultContent, invalidPayloadFile);
+            var bytesToWrite = Encoding.UTF8.GetBytes(payload);
+            if (_errorFileSizeLimitBytes.HasValue)
+            {
+                CleanUpInvalidPayloadFiles(_errorFileSizeLimitBytes.Value - bytesToWrite.Length, _logFolder);
+            }
+            IOFile.WriteAllBytes(invalidPayloadFile, bytesToWrite);
+        }
+
+        static void CleanUpInvalidPayloadFiles(long maxNumberOfBytesToRetain, string logFolder)
+        {
+            try
+            {
+                var candiateFiles = Directory.EnumerateFiles(logFolder, $"{InvalidPayloadFilePrefix}*.json");
+                DeleteOldFiles(maxNumberOfBytesToRetain, candiateFiles);
+            }
+            catch (Exception ex)
+            {
+                SelfLog.WriteLine("Exception thrown while trying to CleanupErrorFiles: {0}", ex);
+            }
+        }
+
+        static IEnumerable<FileInfo> WhereCumulativeSizeLessThanOrEqual(IEnumerable<FileInfo> files, long maxCumulativeSize)
```

Comment:
Since these are fed in newest-first (the `descending` below) will this end up yielding the newer files and break as soon as the old files we want to delete are hit? I may have misread the algorithm, just seems that way on first inspection. Cheers!